ukis_pysat/raster.py

```diff
@@ -103,16 +96,16 @@ def mask_image(self, bbox, crop=True, pad=False, mode="constant", constant_value
             self.dataset = self._pad_to_bbox(bbox, mode, constant_values).open(**self.dataset.meta)
 
         if isinstance(bbox, polygon.Polygon):
-            self.__arr, self.transform = rasterio.mask.mask(self.dataset, [bbox], crop=crop)
+            self.__arr, transform = rasterio.mask.mask(self.dataset, [bbox], crop=crop)
         elif isinstance(bbox, tuple):
-            self.__arr, self.transform = rasterio.mask.mask(self.dataset, [box(*bbox)], crop=crop)
+            self.__arr, transform = rasterio.mask.mask(self.dataset, [box(*bbox)], crop=crop)
         else:
             raise TypeError(f"bbox must be of type tuple or Shapely Polygon")
 
         # update for further processing
         self.dataset.close()  # meta still available when DataSetReader is closed

```

Comment:
could we move dataset.close() and dataset.open() inside the __update_dataset method? this would move everything related to updating into the same method and make it simpler and less error prone to call it.

Reply:
Yes, the reason to not do that is not relevant anymore when we create `meta` outselves and enforce `"GTiff"`.